cli/commands.go

```diff
@@ -195,13 +189,8 @@ func initDaemon(config *conf.MenderConfig,
 
 	controller, err := app.NewMender(config, *mp)
 	if err != nil {
-		// Ignore server certificate error  (See: MEN-2378)
-		if _, ok := errors.Cause(err).(*client.ClientServerCertificateError); ok {
```

Comment:
Why not return an explicit error and catch it in a type-switch like before?

Reply:
`loadServerTrust` cannot return error anymore, it always returns a valid list of certificates, even if it is an empty one. We need the list, so the alternative would be to return both a list *and* an error, which I think goes against common Go patterns.

Reply:
Suggestion (feel free to ignore): 
Maybe a prettier API would be.
`systemCertPool.New()`
Then
`systemCertPool.Load()`
And the load just doesn't return an error?